Filter/StringFilter.php

```diff
@@ -25,28 +25,39 @@ public function filter(ProxyQueryInterface $proxyQuery, $alias, $field, $data)
             return;
         }
 
-        $data['value'] = trim($data['value']);
+        $value = $data['value'] = trim($data['value']);
```

Comment:
but, you are right. Depending on a magic behavior in a chain item before isn't a really good approach.